src/hooks/useOutsideInteraction.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback, MutableRefObject, useEffect, useRef } from 'react';
+
+type EventType =
+  | 'mousedown'
+  | 'mouseup'
+  | 'touchstart'
+  | 'touchend'
+  | 'keydown';
+
+interface UseOutsideInteractionProps {
+  handleOutsideInteraction: () => void;
+  events?: EventType[];
+}
+
+const defaultEvents: EventType[] = ['mousedown', 'touchstart', 'keydown'];
+
+/**
+ * 지정된 element 외부의 interaction을 감지하는 hook
+ * 마우스 클릭, 터치, 키보드 이벤트(Escape 키)를 처리할 수 있습니다.
+ *
+ * @param {Object} props - hook props
+ * @param {() => void} props.handleOutsideInteraction - 외부 interaction 발생 시 실행될 콜백 함수
+ * @param {EventType[]} [props.events] - 감지할 이벤트 (default :['mousedown', 'touchstart', 'keydown'])
+ *
+ * @returns {React.MutableRefObject<HTMLElement | null>} element에 연결할 ref 객체
+ */
+
+const useOutsideInteraction = ({
+  handleOutsideInteraction,
+  events = defaultEvents,
+}: UseOutsideInteractionProps): MutableRefObject<HTMLElement | null> => {
+  const ref = useRef<HTMLElement | null>(null);
```

Comment:
```suggestion
  const ref = useRef<HTMLElement>(null);
```
---
```ts
function useRef<T>(initialValue: T | null): RefObject<T>;
```
[P1] useRef는 기본으로 null을 포함하고 있네요!

Reply:
[f44222e](https://github.com/frontend-opensource-project/use-react-hooks/pull/22/commits/f44222e96bd15aa4c547b2ccae7d48064dbfb5d8) 여기에서 수정되었어요!!